api/src/main/java/org/pmiops/workbench/reporting/ReportingUploadServiceImpl.java

```diff
@@ -117,6 +123,23 @@ public void uploadBatchWorkspace(List<ReportingWorkspace> batch, long captureTim
     issueInsertAllRequest(
         stopwatch, responseMapBuilder, performanceStringBuilder, insertAllRequest);
     log.info(performanceStringBuilder.toString());
+    // Check response and abort the process if any error happens. In this case, verify_snapshot
+    // won't have the 'successful' record, hence we know that is a "bad" dataset.
+    checkResponse(responseMapBuilder.build());
+  }
+
+  /** Uploads a record into verified_snapshot BigQuery table. */
+  @Override
+  public void uploadVerifiedSnapshot(long captureTimestamp) {
+    final InsertAllRequest insertAllRequest =
+        InsertAllRequest.newBuilder(getTableId(VERIFIED_SNAPSHOT_TABLE_NAME))
+            .setIgnoreUnknownValues(false)
+            .addRow(RowToInsert.of(generateInsertId(), getFixedValues(captureTimestamp)))
+            .build();
+    bigQueryService.insertAll(insertAllRequest);
+    // No need to check response, either it success, we get the record, or it fails, no record
```

Comment:
Will we have logs of whether this insertion failed? This may be a reason to check